loadbalancer/algorithm.go

```diff
@@ -16,37 +18,81 @@ type algorithmType int
 const (
 	none algorithmType = iota
 	roundRobinAlgorithm
+	randomAlgorithm
+	consistentHashAlgorithm
 )
 
+var (
+	algorithms = map[algorithmType]initializeAgorithm{
+		roundRobinAlgorithm:     newRoundRobin,
+		randomAlgorithm:         newRandom,
+		consistentHashAlgorithm: newConsistentHash,
+	}
+	defaultAlgorithm = newRoundRobin
+)
+
+func newRoundRobin(endpoints []string) routing.LBAlgorithm {
+	return &roundRobin{
+		index: rand.Intn(len(endpoints)),
+	}
+}
+
 type roundRobin struct {
 	mx    sync.Mutex
 	index int
 }
 
-type algorithmProvider struct{}
+// Apply implements routing.LBAlgorithm with a roundrobin algorithm.
+func (r *roundRobin) Apply(ctx *routing.LBContext) routing.LBEndpoint {
+	r.mx.Lock()
+	defer r.mx.Unlock()
+	r.index = (r.index + 1) % len(ctx.Route.LBEndpoints)
+	return ctx.Route.LBEndpoints[r.index]
+}
+
+type random struct{}
 
-type initializeAgorithm func(endpoints []string) routing.LBAlgorithm
+func newRandom(endpoints []string) routing.LBAlgorithm {
+	return &random{}
+}
 
-var algorithms = map[algorithmType]initializeAgorithm{
-	roundRobinAlgorithm: newRoundRobin,
+// Apply implements routing.LBAlgorithm with a stateless random algorithm.
+func (r *random) Apply(ctx *routing.LBContext) routing.LBEndpoint {
+	return ctx.Route.LBEndpoints[rand.Intn(len(ctx.Route.LBEndpoints))]
```

Comment:
How is this seeded?

Reply:
Seed(1), I will fix it for roundRobin as well

Reply:
done